### PR TITLE
sc-3394 trtl iterator tests

### PR DESCRIPTION
**Pull request by pdeziel**
This adds tests for the two trtl iterator types and fixes some edge cases in the iterators to protect us from panics.

### CODE REVIEW

File: pkg/gds/store/trtl/iterator_test.go [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+
+	// Iterate over an empty namespace
+	iter := trtl.NewTrtlBatchIterator(client, "empty")
+	require.False(iter.Seek([]byte("foo")), "Seek to non-existent key should return false")
+	require.Nil(iter.Key(), "Key should be nil when iterator is exhausted")
+	require.Nil(iter.Value(), "Value should be nil when iterator is exhausted")
+	require.False(iter.Next(), "Next should return false when iterator is exhausted")
+	require.Nil(iter.Value(), "Value should be nil when iterator is exhausted")
+	require.False(iter.Prev(), "Prev should return false when iterator is exhausted")
+	require.Nil(iter.Value(), "Value should be nil when iterator is exhausted")
```
Comment (rebeccabilbro) [inline pkg/gds/store/trtl/iterator_test.go:26]: Perhaps the tests on 22, 24, and 26 should return slightly different errors so that we can more easily distinguish what went wrong if something does?

File: pkg/gds/store/trtl/iterator_test.go [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+
+	// Calling Prev() first
+	iter = trtl.NewTrtlBatchIterator(client, req.Namespace)
+	require.False(iter.Prev(), "Initial Prev should return false")
+	require.Nil(iter.Value(), "Value should be nil when iterator is behind the first key")
+	require.True(iter.Next(), "Next should return true when iterator is behind the first key")
+	require.Equal(req.Key, iter.Key())
+	require.Equal(req.Value, iter.Value())
+	require.Nil(iter.Error())
+
```
Comment (rebeccabilbro) [inline pkg/gds/store/trtl/iterator_test.go:47]: should we `iter.Release()` between lines 46 and 49 just for safety?

File: pkg/gds/store/trtl/iterator_test.go [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+	// Calling Prev() first
+	iter = trtl.NewTrtlStreamingIterator(client, req.Namespace)
+	require.False(iter.Prev(), "Initial Prev should return false")
+	require.Nil(iter.Value(), "Value should be nil when iterator is behind the first key")
+	require.True(iter.Next(), "Next should return true when iterator is behind the first key")
+	require.Equal(req.Key, iter.Key())
+	require.Equal(req.Value, iter.Value())
+	require.Nil(iter.Error())
+
```
Comment (rebeccabilbro) [inline pkg/gds/store/trtl/iterator_test.go:244]: `iter.Release()` here too?

**Review by rebeccabilbro — APPROVED** [review]
Looks good to me @pdeziel -- I just noted a few minor things; otherwise this should be good to merge 👍

File: pkg/gds/store/trtl/iterator_test.go (hunk 1 above)
Comment (pdeziel) [reply pkg/gds/store/trtl/iterator_test.go:26]: Good point, I'll change the wording a bit.
